controllers/openstackmachine_controller.go

```diff
@@ -391,6 +391,9 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 		Address: instanceStatus.Name(),
 	})
 	openStackMachine.Status.Addresses = addresses
+	if openStackMachine.Spec.IdentityRef == nil {
```

Comment:
I'm curious why you need to do that here. We already do it here:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/2210/files#diff-e0a0984c0e26655878e2f94623fbf8cddb4d8a89e3443c1296dfaadc8802838aR578-R583

I'll like to avoid dupplications.

Reply:
function `getOrCreateMachineServer` build an openStackServer object, that's not the purpose of this context so I didn't to use it like that.
Do you think that [this way](https://github.com/MatthieuFin/cluster-api-provider-openstack/commit/9250c73f899906b73aaf9853b3a6446856defc49) is a better approach ?

Reply:
No it's not a better approach I think.
However moving the code to a separated function and use it in both places seems a good idea. We share the logic in case it has to change one day.

Thanks for trying that.

Reply:
Obviously it seems to make sens, I'll take a look on this way.

Reply:
IdentityRef lookup logic was also implmented [here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/0f945e1/pkg/scope/provider.go#L58), so I tried to factorise code to use same implementation at anyplaces with this last [commit](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/2210/commits/dc72b3ce230ee1f5e05c7a56d91d93514dc57c60).